src/artifact_providers/zeus.ts

```diff
@@ -160,25 +159,26 @@ export class ZeusArtifactProvider extends BaseArtifactProvider {
     // the other providers (which overwrite pre-existing, identically-named
     // files within the same commit), for each filename, take the one with the
     // most recent update time
-    const nameToZeusArtifacts = _.groupBy(
-      zeusArtifacts,
-      zeusArtifact => zeusArtifact.name
-    );
-    const dedupedZeusArtifacts = Object.keys(nameToZeusArtifacts).map(
-      zeusArtifactName => {
-        const zeusArtifactObjects = nameToZeusArtifacts[zeusArtifactName];
-        // Sort by the update time
-        const sortedZeusArtifacts = _.sortBy(
-          zeusArtifactObjects,
-          zeusArtifactObject =>
-            Date.parse(zeusArtifactObject.updated_at || '') || 0
-        );
-        return sortedZeusArtifacts[sortedZeusArtifacts.length - 1];
+    zeusArtifacts.sort((a, b) => {
+      if (a.name < b.name) {
+        return -1;
+      } else if (a.name > b.name) {
+        return 1;
+      } else {
+        const aUpdatedAt = Date.parse(a.updated_at ?? '') || 0;
+        const bUpdatedAt = Date.parse(b.updated_at ?? '') || 0;
+        if (aUpdatedAt < bUpdatedAt) {
+          return -1;
+        } else if (aUpdatedAt > bUpdatedAt) {
+          return 1;
+        } else {
+          return 0;
+        }
       }
-    );
+    });
 
-    return dedupedZeusArtifacts.map(zeusArtifact =>
-      this.convertToRemoteArtifact(zeusArtifact)
-    );
+    return zeusArtifacts
+      .filter((artifact, idx, arr) => artifact.name !== arr[idx + 1]?.name)
```

Comment:
This sort + filter is correct, but it is quite hard to read and understand now. Could you make this two for plain old `for` loops for readability, instead? Or, once we drop support for Node 10, we can instead switch to `Object.fromEntries`. 

1. step: get maximum artifact by name
2. step: map and convert

Same in the store impl.

Reply:
Yup, agreed. This is the kind of feedback I was looking for, thanks a lot!